### PR TITLE
Item 7102: Changes for Rlabkey version 2.4.0

### PR DESCRIPTION
**Pull request by cnathe**
#### Rationale
Various client issues and feature requests have come up related to the Rlabkey package and have been prioritized to work on for the next version of the Rlabkey package. 

#### Related Pull Requests
* https://github.com/LabKey/testAutomation/pull/297

#### Changes
* Issue 39955: Fix for labkey.saveBatch issue with using GET instead of POST for query to get assay Id from assay name
* Issue 39955: Add batchPropertyList parameter to labkey.experiment.saveBatch to match behavior from deprecated labkey.saveBatch
* Issue 39965: Add labkey.domain.createIndices helper and indices parameter to labkey.domain.createDesign
* Issue 39966: Add labkey.security.getContainers, labkey.security.createContainer, labkey.security.deleteContainer, labkey.security.moveContainer

### CODE REVIEW

**Comment by cnathe** [comment]
Attaching the versions build on my Windows machine which I will put in svn for TeamCity to run.
[Rlabkey_2.4.0.zip](https://github.com/LabKey/labkey-api-r/files/4427918/Rlabkey_2.4.0.zip)
[Rlabkey_2.4.0.tar.gz](https://github.com/LabKey/labkey-api-r/files/4427919/Rlabkey_2.4.0.tar.gz)

File: Rlabkey/R/labkey.experiment.R [hunk 1]
```diff
@@ -170,7 +170,7 @@ labkey.experiment.saveBatch <- function(baseUrl=NULL, folderPath, assayConfig =
         params = list()
         params$protocolName = protocolName
     }
-    params$batch = list(runs = ensureNestedList(runList))
+    params$batch = c(batchPropertyList, list(runs = ensureNestedList(runList)))
```
Comment (labkey-klum) [inline Rlabkey/R/labkey.experiment.R:173]: should there be a null check before we add the batch properties to the collection?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#Mon Jul 10 08:01:40 PDT 2017
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.3-bin.zip
```
Comment (labkey-klum) [inline gradle/wrapper/gradle-wrapper.properties:3]: yikes, good catch

**Review by labkey-klum — APPROVED** [review]

File: Rlabkey/R/labkey.experiment.R (hunk 1 above)
Comment (cnathe) [reply Rlabkey/R/labkey.experiment.R:173]: Looks like that isn't needed in R. The following both result in the same output:
`list(runs = "test")`
`c(NULL, list(runs = "test"))`
R looks to drop the NULL from the vector creation.

**Pull request merged**